osf/models/preprint.py

```diff
@@ -463,6 +463,21 @@ def create_version(cls, create_from_guid, auth):
                 sentry.log_exception(e)
                 sentry.log_message(f'Contributor was not added to new preprint version due to error: '
                                    f'[preprint={preprint._id}, user={contributor.user._id}]')
+
+        # Add new version record for unregistered contributors
+        for contributor in preprint.contributor_set.filter(user__is_registered=False):
```

Comment:
Is `user__is_registered=False` precise enough? I am raising this question because the error happens in the following `else`.

```python
def csl_name(self, node_id=None):
    # disabled users are set to is_registered = False but have a fullname
    if self.is_registered or self.is_disabled:
        name = self.fullname
    else:
        name = self.get_unclaimed_record(node_id)['name']
```